integrations/airflow/marquez_airflow/extractors/__init__.py

```diff
@@ -124,24 +129,29 @@ def __repr__(self):
 
 
 class StepMetadata:
-    # TODO: Define a common way across extractors to build the
-    # job name for an operator
-    name = None
-    location = None
-    inputs = []
-    outputs = []
-    context = {}
 
-    def __init__(self, name, location=None, inputs=None, outputs=None,
-                 context=None):
+    def __init__(
+            self,
+            name,
+            location=None,
+            inputs: List[Dataset] = None,
+            outputs: List[Dataset] = None,
+            context=None
+    ):
+        # TODO: Define a common way across extractors to build the
+        # job name for an operator
         self.name = name
         self.location = location
-        if inputs:
-            self.inputs = inputs
-        if outputs:
-            self.outputs = outputs
-        if context:
-            self.context = context
+        self.inputs = inputs
+        self.outputs = outputs
+        self.context = context
+
+        if not inputs:
+            self.inputs = []
+        if not outputs:
+            self.outputs = []
```

Comment:
I think we should be able to tell the difference between "we have no inputs" ( [] ) and "I don't know if we have inputs" ( null )

Reply:
Makes sense, but is this distinction meaningful in OpenLineage events?

Reply:
That makes sense, but the question is if the distinction is meaningful for OpenLineage events. If not, then non-nulleable data would be prefered IMO.